lib/OnyxUtils.ts

```diff
@@ -1050,51 +946,47 @@ function broadcastUpdate(key, value, hasChanged, wasRemoved = false) {
         cache.addToAccessedKeys(key);
     }
 
-    return scheduleSubscriberUpdate(key, value, prevValue, (subscriber) => hasChanged || subscriber.initWithStoredValues === false);
+    return scheduleSubscriberUpdate(key, value, prevValue, (subscriber) => hasChanged || subscriber?.initWithStoredValues === false).then(() => undefined);
 }
 
-/**
- * @param {String} key
- * @returns {Boolean}
- */
-function hasPendingMergeForKey(key) {
-    return Boolean(mergeQueue[key]);
+function hasPendingMergeForKey(key: OnyxKey): boolean {
+    return !!mergeQueue[key];
 }
 
 /**
  * Removes a key from storage if the value is null.
  * Otherwise removes all nested null values in objects and returns the object
- * @param {String} key
- * @param {Mixed} value
- * @returns {Mixed} The value without null values and a boolean "wasRemoved", which indicates if the key got removed completely
+ *
+ * @returns The value without null values and a boolean "wasRemoved", which indicates if the key got removed completely
  */
-function removeNullValues(key, value) {
-    if (_.isNull(value)) {
+function removeNullValues(key: OnyxKey, value: OnyxValue<OnyxKey>): {value: Record<string, unknown> | unknown[] | null; wasRemoved: boolean} {
```

Comment:
Could you extract `{value: Record<string, unknown> | unknown[] | null; wasRemoved: boolean}` to a separate type?